File: URI1040.py
notas = input()
A, B, C, D = notas.split(" ")
A = float(A)
B = float(B)
C = float(C)
D = float(D)
media = (A*2+B*3+C*4+D*1)/10
if media >= 7:
    print("Media: {:.1f}".format(media))
    print("Aluno aprovado.")
elif 5 <= media <= 6.9:
    print("Media: {:.1f}".format(media))
    print("Aluno em exame.")
    ne = float(input())
    print("Nota do exame: {:.1f}".format(ne))
    nm = (media + ne)/2
    if nm >= 5:
        print("Aluno aprovado.")
        print("Media final: {:.1f}".format(nm))
    else:
        print("Aluno reprovado.")
        print("Media final {:.1f}".format(nm))
else:
    print("Media: {:.1f}".format(media))
    print("Aluno reprovado.")

File: C/URI1040.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main() {
    
    double n1, n2, n3, n4, n5, media;
    
    scanf("%lf %lf %lf %lf", &n1, &n2, &n3, &n4);
    
    media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1)/10;
    
    printf("Media: %.1lf\n", media);

    if (media >= 7) 
        printf("Aluno aprovado.\n");
    else if (media < 5) {
        printf("Aluno reprovado.\n");
    } else {
        printf("Aluno em exame.\n");
        scanf("%lf", &n5);
        printf("Nota do exame: %.1lf\n", n5);
        if ((n5 + media)/2 >= 5)
            printf("Aluno aprovado.\n");
        else 
            printf("Aluno reprovado.\n");
        printf("Media final: %.1lf\n", (n5+media)/2);
    }   
    
    return 0;

}